docker/utils/utils.py

```diff
@@ -601,6 +602,30 @@ def create_host_config(
             log_config = LogConfig(**log_config)
         host_config['LogConfig'] = log_config
 
+    if cpu_quota and not isinstance(cpu_quota, int):
+        raise TypeError(
+            'Invalid type for cpu_quota param: expected int but'
+            ' found {0}'.format(type(cpu_quota))
+        )
+    elif cpu_quota:
+        if compare_version(version, '1.19') > 0:
+            raise errors.InvalidVersion(
+                'cpu_quota param not supported for API version < 1.19'
+            )
+        host_config['CpuQuota'] = cpu_quota
```

Comment:
Minor thing, but this style of conditional makes me do a double-take - plus we're immediately doing another check in the second branch. I think a single outer check and two inner checks would be more normal.

Also, could we use `version_lt`? I find it much easier to read.

```python
if cpu_quota:
    if not isinstance(cpu_quota, int):
        raise TypeError(...)
    if version_lt(version, '1.19'):
        raise errors.InvalidVersion(...)
    host_config['CpuQuota'] = cpu_quota
```

Same for the `cpu_period` conditional.